src/Microsoft.VisualStudio.Threading/JoinableTask.cs

```diff
@@ -270,16 +270,10 @@ public Task Task
         {
             get
             {
-                using (this.JoinableTaskContext.NoMessagePumpSynchronizationContext.Apply())
```

Comment:
💭 This is the only change I wasn't able to convince myself is necessarily correct. Specifically, I would need to convince myself that the previous code was not using the `lock` here to prevent access to this property prior to a different thread calling `SetWrappedTask`. If you are certain that this isn't actually going to happen, then the change is fine. Otherwise, the safe change seems to be similar to the approach used by other methods - early return without taking the lock if `wrappedTask` is not null.

⚠️ If the order of continuations from `wrappedTask` is important, be aware that this change could allow a caller to register a continuation on `wrappedTask` which executes in front of the one registered by `SetWrappedTask` in the following code:

https://github.com/Microsoft/vs-threading/blob/fd039889435ba4558dea4d309d9f956567355555/src/Microsoft.VisualStudio.Threading/JoinableTask.cs#L695-L710